packages/blockchain-wallet-v4-frontend/src/modals/ShowEthPrivateKey/selectors.js

```diff
@@ -2,11 +2,18 @@ import { selectors } from 'data'
 
 export const getData = (state, props) => {
   const ethKvStoreSelectors = selectors.core.kvStore.ethereum
-  return {
-    balance: props.archived ? 0 : selectors.core.data.ethereum.getBalance(state).getOrElse(0),
-    addr: (props.isLegacy
-      ? ethKvStoreSelectors.getLegacyAccountAddress(state)
-      : ethKvStoreSelectors.getContext(state)).getOrElse(''),
-    priv: state.securityCenter.shownEthPrivKey
+  if (props.isLegacy) {
+    const legacyAccountAddress = ethKvStoreSelectors.getLegacyAccountAddress(state).getOrElse('')
+    return {
+      balance: selectors.core.data.ethereum.getLegacyBalance(state).getOrElse(0),
+      addr: legacyAccountAddress,
+      priv: state.securityCenter.shownEthPrivKey
+    }
+  } else {
+    return {
+      balance: selectors.core.data.ethereum.getBalance(state).getOrElse(0),
+      addr: ethKvStoreSelectors.getContext(state).getOrElse(''),
+      priv: state.securityCenter.shownEthPrivKey
+    }
```

Comment:
Can you please put this first if in a getLegacyData call instead to avoid if/else statements and avoid future confusion?